### PR TITLE
Implement scenario reset functions

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Bit of low hanging fruit, but needed doing at some point.

### CODE REVIEW

File: src/OpenLoco/src/Vehicles/RoutingManager.cpp [hunk 1]
```diff
@@ -51,6 +52,12 @@ namespace OpenLoco::Vehicles::RoutingManager
         std::fill(std::begin(vehRoutingArr), std::end(vehRoutingArr), kRoutingNull);
     }
 
+    // 0x004A8810
+    void resetRoutingTable()
+    {
+        std::fill_n(&routings()[0][0], Limits::kMaxVehicles * Limits::kMaxRoutingsPerVehicle, kRoutingNull);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Vehicles/RoutingManager.cpp:58]: Can't you just use begin and end on this?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Vehicles/RoutingManager.cpp:58]: I think that's not really compatible with how the array is defined. Perhaps I'm missing something, but this doesn't work, anyway:
```cpp
        std::fill(std::begin(routings()), std::end(routings()), kRoutingNull);
```

File: src/OpenLoco/src/Scenario.cpp [hunk 2]
```diff
@@ -159,22 +160,15 @@ namespace OpenLoco::Scenario
     }
 
     // 0x00475988
-    static void sub_475988()
+    static void resetRoadObjects()
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Scenario.cpp:163]: I think this should be moved into ScenarioConstruction.cpp and renames preformed on the existing `resetConstruction` to make it clear one is for road and one is track.

**Review by duncanspumpkin — APPROVED** [review]
All looks fine but maybe we should move that one function and rename the other reset one for that struct.

**Pull request merged**